runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6849,6 +6896,29 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
                (TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) || localColdCompilations))
                doLocalCompilation = true;
 
+            // If this is a remote sync compilation, change it to a local sync compilation.
+            // After the local compilation is completed successfully, a remote async compilation
+            // will be scheduled in compileOnSeparateThread().
+            TR::IlGeneratorMethodDetails & details = entry->getMethodDetails();
+            if (!doLocalCompilation &&
+                !entry->_async &&
+                !entry->_changedToLocalSyncComp &&
```

Comment:
Do we need the test on _changedToLocalSyncComp? I guess the idea of the test to prevent lowering the optLevel for the recompilation, but maybe the _async test already does that.